mayastor/src/bdev/nexus/nexus_child.rs

```diff
@@ -350,8 +350,14 @@ impl NexusChild {
         // The bdev is being removed, so ensure we don't use it again.
         self.bdev = None;
 
+        // Ensure the child is not in the Open state. This guarantees that the
+        // child will be removed from the I/O path when the nexus is
+        // reconfigured.
+        if self.state.load() == ChildState::Open {
```

Comment:
do we need to check if a child is in a rebuild state?

Reply:
Also, I think this is clearer as a match statement without a default state so that no states are forgotten if, say, in the future, new states are introduced.

Reply:
@tiagolobocastro good catch. I've added a check for the out-of-sync state also so that the child doesn't get re-added to the write only channels.

@jonathan-teh good suggestion. I've changed it to a match statement.